server/graphql/v2/input/AccountReferenceInput.js

```diff
@@ -93,3 +94,36 @@ export const fetchAccountWithReference = async (
   }
   return collective;
 };
+
+/**
+ * Retrieves accounts for given ids or slugs
+ *
+ * @param {object} inputs - object containing slugs or ids of the collectives
+ * @param {object} params
+ *    - throwIfMissing: throws an exception if a collective is missing for a given id or slug
+ */
+export const fetchAccountsWithReferences = async (inputs, { throwIfMissing = false }) => {
+  const getSQLConditionFromAccountReferenceInput = inputs => {
+    const conditions = [];
+    inputs.forEach(input => {
+      if (input.id && typeof input.id === 'string') {
+        const id = idDecode(input.id, 'account');
+        conditions.push({ id });
+      } else if (input.legacyId || typeof input.id === 'number') {
+        conditions.push({ id: input.legacyId || input.id });
+      } else if (input.slug) {
+        conditions.push({ slug: input.slug.toLowerCase() });
+      }
+    });
+
+    return conditions;
+  };
+
+  const conditions = getSQLConditionFromAccountReferenceInput(inputs);
+  const accounts = await models.Collective.findAll({ where: { [Op.or]: conditions } });
+  if (accounts.length !== inputs.length && throwIfMissing) {
```

Comment:
:ice_cream: Edge case so feel free to ignore it (but maybe add a comment)?

If you pass `account: [{ slug: "opensource" }, { legacyId: 11004 }]` this will throw (since both input are referring to the same account). We may want our check to look like:

```es6
const accountLoadedForInput = input => accounts.some(account => accountMatchesInput(account, input));
if (throwIfMissing && !inputs.every(accountLoadedForInput)) {
```

But the current code is an ok solution for a first version in my opinion.

Reply:
Thanks for pointing it out. I've done this as well. Let me know how it looks. 😄